plugins/field-colour/src/blocks/colourBlend.ts

```diff
@@ -0,0 +1,296 @@
+ */
+
+import {Block, common as BlocklyCommon} from 'blockly';
+import {
+  JavascriptGenerator,
+  Order as JavascriptOrder,
+} from 'blockly/javascript';
+import {DartGenerator, Order as DartOrder} from 'blockly/dart';
+import {LuaGenerator, Order as LuaOrder} from 'blockly/lua';
+import {PhpGenerator, Order as PhpOrder} from 'blockly/php';
+import {PythonGenerator, Order as PythonOrder} from 'blockly/python';
+import {registerFieldColour} from '../field_colour';
+import {Generators} from './generatorUtils';
+
+/** The name this block is registered under. */
+// eslint-disable-next-line @typescript-eslint/naming-convention
+export const BLOCK_NAME = 'colour_blend';
+
+// Block for blending two colours together.
+const jsonDef = {
+  type: BLOCK_NAME,
+  message0:
+    '%{BKY_COLOUR_BLEND_TITLE} %{BKY_COLOUR_BLEND_COLOUR1} ' +
+    '%1 %{BKY_COLOUR_BLEND_COLOUR2} %2 %{BKY_COLOUR_BLEND_RATIO} %3',
+  args0: [
+    {
+      type: 'input_value',
+      name: 'COLOUR1',
+      check: 'Colour',
+      align: 'RIGHT',
+    },
+    {
+      type: 'input_value',
+      name: 'COLOUR2',
+      check: 'Colour',
+      align: 'RIGHT',
+    },
+    {
+      type: 'input_value',
+      name: 'RATIO',
+      check: 'Number',
+      align: 'RIGHT',
+    },
+  ],
+  output: 'Colour',
+  helpUrl: '%{BKY_COLOUR_BLEND_HELPURL}',
+  style: 'colour_blocks',
+  tooltip: '%{BKY_COLOUR_BLEND_TOOLTIP}',
+};
+
+/**
+ * Javascript block generator function.
+ *
+ * @param block The Block instance to generate code for.
+ * @param generator The JavascriptGenerator calling the function.
+ * @returns A tuple containing the code string and precedence.
+ */
+export function jsGenerator(
+  block: Block,
+  generator: JavascriptGenerator,
+): [string, JavascriptOrder] {
+  // Blend two colours together.
+  const c1 =
+    generator.valueToCode(block, 'COLOUR1', JavascriptOrder.NONE) ||
+    "'#000000'";
+  const c2 =
+    generator.valueToCode(block, 'COLOUR2', JavascriptOrder.NONE) ||
+    "'#000000'";
+  const ratio =
+    generator.valueToCode(block, 'RATIO', JavascriptOrder.NONE) || 0.5;
+  const functionName = generator.provideFunction_(
+    'colourBlend',
+    `
+function ${generator.FUNCTION_NAME_PLACEHOLDER_}(c1, c2, ratio) {
+  ratio = Math.max(Math.min(Number(ratio), 1), 0);
+  var r1 = parseInt(c1.substring(1, 3), 16);
+  var g1 = parseInt(c1.substring(3, 5), 16);
+  var b1 = parseInt(c1.substring(5, 7), 16);
+  var r2 = parseInt(c2.substring(1, 3), 16);
+  var g2 = parseInt(c2.substring(3, 5), 16);
+  var b2 = parseInt(c2.substring(5, 7), 16);
+  var r = Math.round(r1 * (1 - ratio) + r2 * ratio);
+  var g = Math.round(g1 * (1 - ratio) + g2 * ratio);
+  var b = Math.round(b1 * (1 - ratio) + b2 * ratio);
+  r = ('0' + (r || 0).toString(16)).slice(-2);
+  g = ('0' + (g || 0).toString(16)).slice(-2);
+  b = ('0' + (b || 0).toString(16)).slice(-2);
+  return '#' + r + g + b;
+}
+`,
+  );
+  const code = functionName + '(' + c1 + ', ' + c2 + ', ' + ratio + ')';
```

Comment:
```suggestion
  const code = `${functionName}(${c1}, ${c2}, ${ratio})`;
```

Here and similarly below.

Reply:
Tracked in #2198 for follow-up.